api/main.go

```diff
@@ -1,26 +1,36 @@
 package main
 
 import (
+	"context"
+
 	_ "github.com/chanzuckerberg/happy-api/docs" // import API docs
 	"github.com/chanzuckerberg/happy-api/pkg/api"
+	"github.com/chanzuckerberg/happy-api/pkg/request"
 	"github.com/chanzuckerberg/happy-api/pkg/setup"
 	"github.com/sirupsen/logrus"
 	"gopkg.in/yaml.v3"
 )
 
 func exec() error {
-	config, err := setup.GetConfiguration()
+	cfg, err := setup.GetConfiguration()
 	if err != nil {
 		return err
 	}
 
-	m, err := yaml.Marshal(config)
+	m, err := yaml.Marshal(cfg)
 	if err != nil {
 		return err
 	}
 	logrus.Info("Running with configuration:\n", string(m))
 
-	app := api.MakeApp(config)
+	cfg.Auth.Verifier, err = request.MakeOIDCVerifier(context.Background(), cfg.Auth.IssuerURL, cfg.Auth.ClientID)
+	if err != nil {
+		return err
+	}
+	app, err := api.MakeApp(cfg)
```

Comment:
I opted for this approach because all the other approaches were dangerously close to mixing production and test code together. This allows us to still use the configuration object, we just alter it a little bit before it goes into the makeapp call.

Reply:
There is also less "magic" string, which I think it hard to maintain. Its confusing unless we document it really well